packages/std/src/math/uint512.rs

```diff
@@ -508,6 +476,14 @@ impl Rem for Uint512 {
 }
 forward_ref_binop!(impl Rem, rem for Uint512, Uint512);
 
+impl Not for Uint512 {
+    type Output = Self;
+
+    fn not(self) -> Self::Output {
+        Self(!self.0)
+    }
+}
+
```

Comment:
Yes, we should do that, but I decided to keep that as a [separate issue](https://github.com/CosmWasm/cosmwasm/issues/1754) outside of this PR.
This one slipped in here because I needed it during the initial prototyping.